components/layout_2020/fragment_tree/fragment.rs

```diff
@@ -70,6 +71,9 @@ pub(crate) struct TextFragment {
 
     /// Extra space to add for each justification opportunity.
     pub justification_adjustment: Au,
+    pub insertion_point: Option<ByteIndex>,
```

Comment:
We can't have caret and range selections at the same time. Can we merge them into an enum?

Ditto for all other places where they appear together.

This will eliminate any possibility we paint both caret and range.